src/utils/helpers.tsx

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+
+interface ILinkProps {
```

Comment:
- `interface ICustomLinkProperties`
- `readonly` all properties
- `content` is not a thing. Use [`children`](https://reactjs.org/docs/jsx-in-depth.html#children-in-jsx)
- `children` wants to be of type `ReactNode` instead of `string`. Try: `<CustomLink>Click <strong>me</strong></CustomLink>` for instance
- `classes` should be `className` in order to follow on with already established principles of react
- `className` should be optional: `readonly className?: string;`

Reply:
Better yet, just do this:

```tsx
function CustomLink(props: React.DetailedHTMLProps<
    React.AnchorHTMLAttributes<HTMLAnchorElement>,
    HTMLAnchorElement>): ReactElement {
  return <a {...props} onChange={e => {
    e.preventDefault();
    history.push(props.href);
    return false;
  }></a>;
}
```

that means all of the properties of an anchor element will now be in your element, with you overriding the onChange call.

Reply:
@paroxpI Now I have something like this
<img width="1020" alt="Zrzut ekranu 2020-01-18 o 15 03 26" src="https://user-images.githubusercontent.com/17521804/72664933-bd17fc00-3a03-11ea-85ec-e15eb4feca18.png">
And second question, should I use `const history = useHistory();` in helper function or in each component separately?

Reply:
```diff
diff --git a/.eslintrc b/.eslintrc
index e6a6a62..232ade3 100644
--- a/.eslintrc
+++ b/.eslintrc
@@ -58,7 +58,6 @@
     "functional/no-conditional-statement": "off",
     "functional/no-expression-statement": "off",
     "functional/prefer-type-literal": "off",
-    "indent": ["error", 2],
     "jsx-quotes": ["error", "prefer-double"],
     "linebreak-style": ["error", "unix"],
     "newline-before-return": "error",
diff --git a/src/utils/helpers.tsx b/src/utils/helpers.tsx
index 56b4da0..f0fca56 100644
--- a/src/utils/helpers.tsx
+++ b/src/utils/helpers.tsx
@@ -1,17 +1,31 @@
-import React from 'react';
+import React, { ReactElement } from 'react';
 import { useHistory } from 'react-router-dom';
 
-interface ILinkProps {
-  href: string;
-  content: string;
-  classes: string;
+interface ICustomLinkProperties
+  extends React.DetailedHTMLProps<
+    React.AnchorHTMLAttributes<HTMLAnchorElement>,
+    HTMLAnchorElement
+  > {
+  readonly href: string;
 }
 
-export function CustomLink(props: ILinkProps) {
+type IrecordHistory = (event: React.FormEvent<HTMLAnchorElement>) => boolean;
+
+function recordHistory(href: string): IrecordHistory {
   const history = useHistory();
 
-  return <a href={props.href} className={props.classes} onClick={e => {
+  return e => {
     e.preventDefault();
-    history.push(props.href);
-  }}>{props.content}</a>;
+    history.push(href);
+
+    return false;
+  };
+}
+
+export function CustomLink(props: ICustomLinkProperties): ReactElement {
+  return (
+    <a {...props} onChange={recordHistory(props.href)}>
+      {props.children}
+    </a>
+  );
 }

```